app/featureset/config.go

```diff
@@ -0,0 +1,117 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package featureset
+
+import (
+	"context"
+	"math"
+	"strings"
+	"testing"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/app/z"
+)
+
+const (
+	enable  status = math.MaxInt
+	disable status = 0
+)
+
+// Config configures the feature set package.
+type Config struct {
+	// MinStatus defines the minimum enabled status.
+	MinStatus string
+	// Enabled overrides min status and enables a list of features.
+	Enabled []string
+	// Disabled overrides min status and disables a list of features.
+	Disabled []string
+}
+
+// DefaultConfig returns the default config enabling only stable features.
+func DefaultConfig() Config {
+	return Config{
+		MinStatus: statusStable.String(),
+	}
+}
+
+// Init initialises the global feature set state.
+func Init(ctx context.Context, config Config) error {
```

Comment:
so basically minStatus is very important and enabled and disabled are not that important it seems

Reply:
yup, `minStatus` enables/disables different sets of features. While `Enable/Disable` is to override individual features